python/ray/llm/_internal/serve/deployments/utils/node_initialization_utils.py

```diff
@@ -143,9 +143,12 @@ def _initialize_local_node(
     if not isinstance(local_path, str) or not os.path.exists(local_path):
         logger.info(f"Downloading the tokenizer for {engine_config.actual_hf_model_id}")
 
-    # TODO (Kourosh): commented out since for Mistral models that don't support
-    # tekken this code does not work (e.g. mistralai/Devstral-Small-2505)
-    # _ = transformers.AutoTokenizer.from_pretrained(
-    #     engine_config.actual_hf_model_id,
-    #     trust_remote_code=engine_config.trust_remote_code,
-    # )
+    if engine_config.engine_kwargs.get("tokenizer_mode", None) == "mistral":
```

Comment:
is there a more generic thing we can do here, than just filtering based on the name?

Also meta level question: Do we even need this line now that everything is delegated to vllm?

Reply:
yeah. I think it's needed still (if multiple replicas of the same model are to be downloaded on the same node there would be a contention) so you need to handle that at the orchestrator level. Now the question is can this now be simplified / made more robust given the learnings we had? 

Also the important part is really the model weights not the tokenizer / processor. So those things can still be delegated. Maybe we should upstream some locking mechanism to vllm so we don't have to deal with this here at all either.

Reply:
Can we just call get_tokenizer from vllm for this part?